crates/bevy_text/src/style.rs

```diff
@@ -0,0 +1,674 @@
+use crate::*;
+use bevy_asset::AssetId;
+use bevy_asset::Handle;
+use bevy_color::Color;
+use bevy_derive::Deref;
+use bevy_derive::DerefMut;
+use bevy_ecs::component::Component;
+use bevy_ecs::prelude::*;
+use bevy_ecs::relationship::Relationship;
+use bevy_reflect::Reflect;
+
+/// Text style
+#[derive(Clone, PartialEq)]
+pub struct TextStyle {
+    /// The font used by a text entity when neither it nor any ancestor has a [`FontFace`] component.
+    pub font: Handle<Font>,
+    /// Default value
+    pub font_size: FontSize,
+    /// The color used by a text entity when neither it nor any ancestor has a [`TextColor`] component.
+    pub color: Color,
+    /// Default value
+    pub font_smoothing: FontSmoothing,
+    /// Default value
+    pub line_height: LineHeight,
+}
+
+impl TextStyle {
+    /// Returns the text style as a bundle of components
+    pub fn bundle(&self) -> impl Bundle {
+        (
+            FontFace(self.font.clone()),
+            self.font_size,
+            TextColor(self.color),
+            self.font_smoothing,
+            self.line_height,
+        )
+    }
+}
+
+impl Default for TextStyle {
+    fn default() -> Self {
+        Self {
+            font: Default::default(),
+            color: Color::WHITE,
+            font_smoothing: FontSmoothing::default(),
+            line_height: LineHeight::default(),
+            font_size: FontSize::default(),
+        }
+    }
+}
+
+/// Fallback text style used if a text entity and all its ancestors lack text styling components.
+#[derive(Resource, Default, Clone, Deref, DerefMut)]
+pub struct DefaultTextStyle(pub TextStyle);
+
+/// Wrapper used to differentiate propagated text style compoonents
+#[derive(Debug, Component, Clone, PartialEq, Reflect)]
+#[reflect(Component, Clone, PartialEq)]
+pub struct InheritedTextStyle<S: Clone + PartialEq>(pub S);
```

Comment:
I think we are using the word `Style` inconsistently here, does it refer to a single styling property or several? Maybe calling this `InheritedTextProperty` might be clearer.